Sources/Private/Experimental/Layers/ShapeLayer.swift

```diff
@@ -30,15 +30,45 @@ final class ShapeLayer: BaseCompositionLayer {
       let itemsInGroup = group.items.map { ShapeItemLayer.Item(item: $0, parentGroup: group) }
         + otherItems
 
-      // Each `ShapeItemLayer` can only render a single `ShapeItem` that draws a `GGPath`,
-      // so we create a separate `ShapeItemLayer` for each path-drawing `ShapeItem`.
       let pathDrawingItemsInGroup = itemsInGroup.filter { $0.item.drawsCGPath }
       let otherItemsInGroup = itemsInGroup.filter { !$0.item.drawsCGPath }
 
-      for pathDrawingItem in pathDrawingItemsInGroup {
-        let sublayer = ShapeItemLayer(shape: pathDrawingItem, otherItems: otherItemsInGroup)
+      // If all of the path-drawing `ShapeItem`s have keyframes with the same timing information,
+      // we can combine the `[KeyframeGroup<BezierPath>]` (which have to animate in separate layers)
+      // into a single `KeyframeGroup<[BezierPath]>`, which can be combined into a single CGPath animation.
+      //
+      // This is how Groups with multiple path-drawing items are supposed to be rendered,
+      // because combing multiple paths into a single `CGPath` (instead of render them in separate layers)
```

Comment:
```suggestion
      // because combing multiple paths into a single `CGPath` (instead of rendering them in separate layers)
```

Reply:
fixing this in #1507 to prevent merge conflicts, ty for the close eye